middleware/compress.go

```diff
@@ -144,3 +156,32 @@ func gzipCompressPool(config GzipConfig) sync.Pool {
 		},
 	}
 }
+
+// removeOneHeaderValue manipulates the []string value of one header to
+// remove an unwanted item from it.
+func removeOneHeaderValue(header http.Header, key, unwanted string) {
+	key = textproto.CanonicalMIMEHeaderKey(key)
```

Comment:
`textproto.CanonicalMIMEHeaderKey(key)` is not necessary as all keys in `http.Request.Header` as already in canonical mime format and you pass in value with canonical format. 

You could rename it to `removeAcceptEndcodingFromVary()` or something and pull in all those involved header arguments. This adds extra layer of mental gymnastics by being so generic.

Reply:
I considered not including line 163, but decided that this way it's less prone to arbitrary changes at the call site(s).

Also, I called it `removeAcceptEndcodingFromVary()` initially, but realised it's easy to generalise. Perhaps it's a shame the core API has `Add()` but no corresponding `Remove()`; it's what this function does of course.

Reply:
I do not think we need to generalise it here. It adds unnecessary cognitive load. If you do not know that method signature and do not know what vary and etc is then `removeOneHeaderValue(res.Header(), echo.HeaderVary, echo.HeaderAcceptEncoding)` does not tell you much as you would still need to navigate to that method implementation and see what is actually done.
Now when you start from `removeOneHeaderValue` method you are missing the context what for this method exisist in this middleware file. So now you need to search where it is used and what are the parameters. 

all of this is unnecessary in scope of this middleware.  I would somewhat understood what the intention was if it would be added to some other files like `utils.go` or `middleware.go` etc. This file is not the place where to fix standard library deficiencies.  Go philosophy and code is at time banal - keep things simple. even if it does not look super eloquent (in sense of being or not being generic). 

So I till suggest to refactor this method to `removeAcceptEndcodingFromVary(header http.Header)` and use `echo.HeaderVary` `echo.HeaderAcceptEncoding` directly without having them as variables.